src/traces/waterfall/plot.js

```diff
@@ -60,11 +60,6 @@ function plotConnectors(gd, plotinfo, cdModule, traceLayer) {
             var x2, y2;
             var x3, y3;
 
-            var delta = 0;
-            if(i + 1 < len && Array.isArray(trace.offset)) {
-                delta -= trace.offset[i + 1] - trace.offset[i];
```

Comment:
Hmm. Where is `offset` taken in consideration? Do we have a mock with arrayOk `trace.offset` settings?

Reply:
Yes we have: https://github.com/plotly/plotly.js/blob/master/test/image/mocks/waterfall_custom.json

Also here is [a codepen](https://codepen.io/MojtabaSamimi/pen/ZZxoMa?editors=0010) using this branch.

Reply:
Awesome!